src/main/java/com/kodilla/ecommercee/controller/GroupController.java

```diff
@@ -1,31 +1,41 @@
 package com.kodilla.ecommercee.controller;
 
 import com.kodilla.ecommercee.domain.dto.GroupDto;
+import com.kodilla.ecommercee.exception.NotFoundException;
+import com.kodilla.ecommercee.mapper.GroupMapper;
+import com.kodilla.ecommercee.service.GroupService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.MediaType;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @RequestMapping("api/v1/ecommercee/groups")
 public class GroupController {
+    @Autowired
+    GroupService service;
+
+    @Autowired
+    GroupMapper mapper;
 
     @GetMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
     public List<GroupDto> getAllGroups() {
-        return new ArrayList<GroupDto>();
+        return mapper.mapToGroupDtoList(service.getAllGroups());
     }
 
     @GetMapping(path = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public GroupDto getGroup(@PathVariable Long groupId) {
-        return new GroupDto();
+    public GroupDto getGroup(@PathVariable Long groupId) throws NotFoundException {
+        return mapper.mapToGroupDto(service.getGroup(groupId).orElseThrow(NotFoundException::new));
     }
 
     @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
     public void addGroup(@RequestBody GroupDto groupDto) {
+        service.saveGroup(mapper.mapToGroup(groupDto));
     }
 
     @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public void updateGroup(@PathVariable Long id, @RequestBody(required = false) GroupDto groupDto) {
+    public GroupDto updateGroup(@PathVariable Long id, @RequestBody(required = false) GroupDto groupDto) {
+        return mapper.mapToGroupDto(service.saveGroup(mapper.mapToGroup(groupDto)));

```

Comment:
Check whether it gas null id
```suggestion
   log.debug("REST request to save Grup : {}", groupDto);
        if (groupDto.getId() != null) {
            return ResponseEntity.badRequest().body("Group with given id already exists");
        }
        GroupDto result = service.save(groupDto);
        return ResponseEntity.created(new URI("/api/v1/ecommercee/groups/" + result.getId()))
                .body(result);
```